pkg/kubernetes/provider_single.go

```diff
@@ -26,12 +27,23 @@ func init() {
 // newSingleClusterProvider creates a provider that manages a single cluster.
 // Validates that the manager is in-cluster when the in-cluster strategy is used.
 func newSingleClusterProvider(strategy string) ProviderFactory {
-	return func(m *Manager, cfg *config.StaticConfig) (Provider, error) {
+	return func(cfg *config.StaticConfig) (Provider, error) {
```

Comment:
should the comment for the func be updated ? It states that the manager is being validated, but I guess that's more based on the old behavior where a manager is passed in. 

Now it is handled internally